### PR TITLE
Derived data test

### PR DESCRIPTION
**Pull request by Cyberbeni**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

https://github.com/fastlane/fastlane/issues/11642

### Description

Changed `Scan::DetectValues` test to include workspace relative DerivedData location via `WorkspaceSettings.xcsettings` and expect project path as derived data path

### CODE REVIEW

File: scan/spec/detect_values_spec.rb [hunk 1]
```diff
@@ -3,13 +3,15 @@
     describe 'Xcode config handling' do
       before do
         options = { project: "./scan/examples/standard/app.xcodeproj" }
+        FileUtils.mkdir_p("./scan/examples/standard/app.xcodeproj/project.xcworkspace/xcuserdata/#{ENV['USER']}.xcuserdatad/")
+        FileUtils.copy("./scan/examples/standard/WorkspaceSettings.xcsettings", "./scan/examples/standard/app.xcodeproj/project.xcworkspace/xcuserdata/#{ENV['USER']}.xcuserdatad/WorkspaceSettings.xcsettings")
```
Comment (joshdholtz) [inline scan/spec/detect_values_spec.rb]: Is this `ENV['USER`]` needed in these tests? Is it possible for us to put a static path there instead?

**Review by joshdholtz — COMMENTED** [review]
One question!

File: scan/spec/detect_values_spec.rb (hunk 1 above)
Comment (Cyberbeni) [reply scan/spec/detect_values_spec.rb]: These settings can't be in the shared xcsettings, so we need to use the current user.

Comment (joshdholtz) [reply scan/spec/detect_values_spec.rb]: 👌

**Review by joshdholtz — APPROVED** [review]
💪

**Pull request merged**